types/src/oauth/oauth_api.ts

```diff
@@ -152,7 +153,7 @@ export class OAuthAPI {
     provider: CredentialsProvider;
     userId: string;
     workspaceId: string;
-    credentials: ConnectionCredentials;
+    credentials: ConnectionCredentials | ApiKeyCredentials;
```

Comment:
Have you tested this ? My guess is this will be rejected by oauth as currently it only accepts Snowflake's format.

Reply:
I believe @fontanierh is correct, you probably need to update oauth (you'll find some tests there btw, it makes it easy to test by adding some for your new credential format).

Reply:
Yep, definitely overlooked. Pushed the changes but I need to test a bit more